src/mono/configure.ac

```diff
@@ -6795,6 +6795,41 @@ AC_COMPILE_IFELSE(
 		AC_MSG_RESULT(no)
 	])
 
+# for icu shim
+if test x$with_core = xonly; then
+	ICU_SHIM_PATH=../../../libraries/Native/Unix/System.Globalization.Native
+	if test x$target_osx = xyes; then
+		ORIG_CPPFLAGS=$CPPFLAGS
+		# adding icu path to pkg_config_path
+		PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/lib/pkgconfig:/usr/local/opt/icu4c/lib/pkgconfig
+		export PKG_CONFIG_PATH
+		CPPFLAGS="`pkg-config --cflags-only-I icu-uc`"
+		AC_CHECK_LIB(icucore, ucol_open, [],
+			[AC_MSG_ERROR([Cannot find libicucore, skipping build for System.Globalization.Native. .NET globalization is not expected to function.])])
+		AC_DEFINE(ICU_LIBS, -licucore, [adding -licucore])
+		AC_CHECK_HEADER(unicode/utypes.h, [have_sys_icu=yes], [have_sys_icu=no])
+		if test x$have_sys_icu = xyes; then
+			ICU_CFLAGS="$CPPFLAGS -DOSX_ICU_LIBRARY_PATH=AS_ESCAPE(\"/usr/lib/libicucore.dylib\", '\"') -DTARGET_UNIX -DU_DISABLE_RENAMING -Wno-reserved-id-macro -Wno-documentation -Wno-documentation-unknown-command -Wno-switch-enum -Wno-covered-switch-default -Wno-covered-switch-default -Wno-extra-semi-stmt -Wno-unknown-warning-option -Wno-deprecated-declarations"
```

Comment:
Where are these warnings coming from? Not saying they look bad, just wondering why this specific list.

Reply:
> @thaystg I see that the native source files are using on Windows are coming in from Unix folders. Is this how it should work even on Windows or does CoreClr do something different in this area, not using ICU on Windows but falling back to OS specific API's, or is the ambition to always use ICU on all platforms going forward?

@safern told me to use from Unix, they are doing the same on CoreClr, but I think he is planning to move it to another folder.